Jellyfin.Server/Program.cs

```diff
@@ -143,87 +143,95 @@ private static async Task StartApp(StartupOptions options)
+        /// for everything else the XDG approach is followed:
+        /// https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
+        /// </summary>
+        /// <param name="options"></param>
+        /// <returns>ServerApplicationPaths</returns>
         private static ServerApplicationPaths CreateApplicationPaths(StartupOptions options)
         {
-            string programDataPath = Environment.GetEnvironmentVariable("JELLYFIN_DATA_PATH");
-            if (string.IsNullOrEmpty(programDataPath))
+            string dataDir;
+            string configDir;
+            string logDir;
+
+            // dataDir
+            // IF      --datadir
+            // ELSE IF $JELLYFIN_DATA_PATH
+            // ELSE IF windows, use <%APPDATA%>/jellyfin
+            // ELSE IF $XDG_DATA_HOME then use $XDG_DATA_HOME/jellyfin
+            // ELSE    use $HOME/.local/share/jellyfin
+            if (options.DataDir != null)
             {
-                if (options.DataDir != null)
-                {
-                    programDataPath = options.DataDir;
-                }
-                else
-                {
-                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-                    {
-                        programDataPath = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
-                    }
-                    else
-                    {
-                        // $XDG_DATA_HOME defines the base directory relative to which user specific data files should be stored.
-                        programDataPath = Environment.GetEnvironmentVariable("XDG_DATA_HOME");
-                        // If $XDG_DATA_HOME is either not set or empty, $HOME/.local/share should be used.
-                        if (string.IsNullOrEmpty(programDataPath))
-                        {
-                            programDataPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".local", "share");
-                        }
-                    }
-
-                    programDataPath = Path.Combine(programDataPath, "jellyfin");
-                }
+                 dataDir = options.DataDir;
             }
-
-            if (string.IsNullOrEmpty(programDataPath))
+            else if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("JELLYFIN_DATA_PATH")))
             {
-                Console.WriteLine("Cannot continue without path to program data folder (try -programdata)");
-                Environment.Exit(1);
+                dataDir = Environment.GetEnvironmentVariable("JELLYFIN_DATA_PATH");
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                dataDir = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "jellyfin");
+            }
+            else if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("XDG_DATA_HOME")))
```

Comment:
Use a variable

Reply:
I see no benefit, other than breaking the nice logic flow (if, elseif, elseif, elseif, else)..  Copying to variable first will be more expensive than this simple null check on byte[0].

Reply:
Doubt what?  Assignment + check is not greater than just check?

Reply:
Perhaps I was expecting too much from the compiler optermisations..  I'll re-write to be more efficient.  Same problem in datadir and logdir..